### ISSUE TITLE
Constructing of histogram takes a long time

### ISSUE THREAD

**Issue by soichih** [brainlife/validator-neuro-track#8]
The construction of histogram sometimes take more than several minutes (and the validation job times out)

```
    lengths = np.array(list(length(streamlines)))
    hist, bin_edges = np.histogram(lengths, bins='auto')
    bin_edges = np.round(bin_edges, decimals=2)
```

I believe the use of python list/length functions are too slow for our use case. Is there way to speed this up by using numpy native methods? @giulia-berto

**Comment by giulia-berto** [comment]
Why don't we try resampling the streamlines before computing their lengths? In this case I assume we don't have to be super accurate, so even nb_points=12 should work.
```
from dipy.tracking.streamline import set_number_of_points
streamlines=set_number_of_points(streamlines, 12)
```

**Comment by soichih** [comment]
We are already sampling streams to render the thumbnail, so we can just use that? 

How do you account of decreased number of stream counts though? If we sample 10% of the streamline, do we just multiply the count by 10 - assuming that samples are randomly distributed?

**Comment by giulia-berto** [comment]
no, I meant resampling the number of points of the streamlines (e.g. 12 equally-spaced points per streamline). The total number of streamlines should remain the same.
